### ISSUE TITLE
Move GridField "Import CSV" into modal, with better UX

### ISSUE THREAD

**Issue by clarkepaul** [silverstripe/silverstripe-framework#6431]
A partial fix for this issue is included in this PR https://github.com/silverstripe/silverstripe-framework/pull/6354 but it doesn't resolve all the experience completely.

Having it in the modal has improved the usability somewhat since it was sitting loose under gridfields but still room for improvements.

Current implementation:
![image](https://cloud.githubusercontent.com/assets/555033/21337839/a5c915f6-c6d5-11e6-9122-7536eb05eb4b.png)

Proposed UX and designs improvements:
https://projects.invisionapp.com/boards/7XX68PM4TPCR/

**Comment by sminnee** [comment]
@clarkepaul can you elaborate on what "needs some love" might mean? :-)

**Comment by clarkepaul** [comment]
This UI doesn't really make sense if you go through the steps to import. The "Import from CSV" button looks like it does the same action as "Choose file"—can we do away with the "Import from CSV" button completely? This could also be a candidate for a drop zone. Once you have imported from CSV is there any way to undo (doubtful but thought it was worth clarifying in the off chance)?

The "Done" action is very out of place compared the rest of the CMS UI, so it might need some thought as to what benefit it provides over providing a success message and closing the modal automatically (it is a duplicate action as the X button as well).

It would be good to review the text/labels to see if we can improve them in any way also.

**Comment by clarkepaul** [comment]
This issue is marked as "Type/UX" as it needs some time from a designer or dev willing to provide a nicer solution than the current experience (prior to development).

**Comment by jamessilverstripe** [comment]
![ss import csv modal - initial](https://user-images.githubusercontent.com/29618750/27415739-c43735dc-575c-11e7-81eb-8ec87a6e9388.png)
![ss import csv modal - uploaded](https://user-images.githubusercontent.com/29618750/27415740-c44e18c4-575c-11e7-9321-75bd87ed326b.png)
![ss import csv modal - done](https://user-images.githubusercontent.com/29618750/27415738-c4034574-575c-11e7-9a8e-0eb4beeb1da0.png)

"Add from files" would only show CSV files as selectable

**Comment by clarkepaul** [comment]
UX completed as above, ready for sizing and development.

**Comment by clarkepaul** [comment]
Comment from @lerni 
> CSV-Import is also hard to find. And if you import, the Success Message like "Imported 20 records." gets hidden in the collapsed Search, since "Import from CSV" collapses Search Panel.

**Comment by lerni** [comment]
@clarkepaul cool, this looks good. This is 4.x only?

**Comment by dhensby** [comment]
1. We're missing a design where the "show advanced usage" is exapanded. I don't think that the existing text makes it very clear that there's an expanded view and that design should be looked into again.
2. Some feedback of what was actually imported (added/deleted/changed/failed) would be good from a users point of view.

**Comment by chillu** [comment]
I've moved James' designs into the ticket description, and keeping Paul's intermediary designs in this comment for later reference:

![pasted_image_20_12_16__4_59_pm](https://cloud.githubusercontent.com/assets/555033/21337846/b8f20750-c6d5-11e6-8501-1600deb7e83f.png)

![image](https://cloud.githubusercontent.com/assets/555033/21337823/8a7bb060-c6d5-11e6-8aee-69faf3365c65.png)

**Comment by clarkepaul** [comment]
@lerni yes only for SS4 unfortunately.

@dhensby I've taken note of your suggestions and will see if we can get them added in there too so they get incorporated.

**Comment by jamessilverstripe** [comment]
Latest designs here: https://projects.invisionapp.com/boards/7XX68PM4TPCR/

@dhensby ^

**Comment by chillu** [comment]
The whole feature is broken at the moment, let's fix that first: https://github.com/silverstripe/silverstripe-framework/issues/7162

**Comment by chillu** [comment]
@flamerohr How much work do you think this is? We need the feature working, but CSV import isn't a highly used feature - so I'm keen to reign in the effort here (for 4.0 release at least)

**Comment by flamerohr** [comment]
@chillu I don't think it's that much work involved, it's a regression from the bootstrap4 upgrade from alpha2 to beta1 (e.g. class `in` has renamed to `show` for all showing to hiding effects)

**Comment by chillu** [comment]
Sweet, then let's go with the designs!

**Comment by tractorcow** [comment]
Fix at https://github.com/silverstripe/silverstripe-admin/pull/210

**Comment by dhensby** [comment]
did https://github.com/silverstripe/silverstripe-admin/pull/210 not fix the issue?

**Comment by flamerohr** [comment]
No, the pull request only fixed it to be useable, but the design specs have not been actioned

**Comment by tractorcow** [comment]
Updated the title to reflect this.

**Comment by chillu** [comment]
OK, doesn't look broken any more. The new design is nice, but not essential.

**Comment by clarkepaul** [comment]
@newleeland can you take a look at this one and see if its design is still using the most recent UI for development.

**Comment by newleeland** [comment]
Just some minor updates

![screen shot 2017-12-21 at 3 30 27 pm](https://user-images.githubusercontent.com/8085295/34237917-02d0bf3e-e664-11e7-8cbf-2d19885675d6.png)

**Comment by dhensby** [comment]
I think the success import needs to be able to show "x imported, y deleted, z update"?

**Comment by flamerohr** [comment]
I wasn't aware of csv import having a delete function?

**Comment by dhensby** [comment]
> I wasn't aware of csv import having a delete function?

I believe it's hidden behind the "show advanced options" link... you can truncate the table before import

**Comment by flamerohr** [comment]
oh right, the "clear everything before import starts" checkbox :D forgot about that

**Comment by mikeyc7m** [comment]
the csv help dropdown is too cluttered. it needs to list the csv column and its database equivalent (using field label, not field name.) Note that uPPERcASE matters on those csv columns!

also the "Replace data" checkbox is a disaster. my client just obliterated his uat environment (thank god he did it before launch). i've renamed it "Destroy all existing data first" to sound scarier but i was unable to remove it without breaking the form (i might have to hide it with css.)

**Comment by mikeyc7m** [comment]
also some other weird behaviour, on a large import it seems to hang and eventually the browser goes white and assumes the post failed. refresh the page and it appears it has finished, but only a few records made it. refresh again and the number of records has increased - it is still importing!

what this needs is to submit via ajax and replace the form with a "processing" animation, maybe even a message that says "come back later to see the results".
